src/app/ui/color_bar.cpp

```diff
@@ -1182,9 +1183,16 @@ void ColorBar::updateCurrentSpritePalette(const char* operationName)
           cmd->execute(UIContext::instance());
         }
         else {
-          Tx tx(writer.context(), operationName, ModifyDocument);
-          tx(cmd);
-          tx.commit();
+          std::unique_ptr<Tx> tx;
+          try {
+            tx = std::unique_ptr<Tx>(new Tx(writer.context(), operationName, ModifyDocument));
```

Comment:
There is no need to create a pointer for Tx (it can be just a `Tx` as it was, we prefer value types instead of pointers).

Reply:
Thanks for the reply!

But if we just use a `Tx`, as the below code:
```c++
try {
  Tx tx(writer.context(), operationName, ModifyDocument);
  tx(cmd);
  tx.commit();
} catch (...) {
  delete cmd;
  throw;
}
```
There might be a double free on `cmd` if `tx(cmd)` throws an exception. (`Transaction::execute` will free `cmd`)

Using a pointer is a bit weird but I'm not sure if there are any other ways to do this.

Reply:
Oh, I'm sorry. I'll have a look at the CLA.

Reply:
The best option is to avoid try/catch at all using a std::unique_ptr. I've pushed a new commit with a fix for this situation: https://github.com/aseprite/aseprite/commit/89f8e6f275869d5c5f7a27b1f5d8676526bd2a3e